### PR TITLE
[CARBONDATA-3017] Map DDL Support

### PR DESCRIPTION
**Pull request by manishnalla1994**
Support Create DDL for Map type.

 - [ ] Any interfaces changed?
 
 - [ ] Any backward compatibility impacted?
 
 - [ ] Document update required?

 - [x] Testing done
       
 - [ ] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA.

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/810/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.3.1, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/9075/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/1007/

**Comment by manishnalla1994** [comment]
Retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/818/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/1015/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.3.1, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/9083/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/1057/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.3.1, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/9127/

**Comment by manishnalla1994** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/868/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/1066/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.3.1, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/9134/

File: integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/createTable/TestCreateDDLForComplexMapType.scala [hunk 1]
```diff
@@ -0,0 +1,330 @@
...
+         | OPTIONS('header'='false')
+       """.stripMargin)
+    checkAnswer(sql("select * from carbon"), Seq(
+      Row(Map(1 -> "Nalla", 2 -> "Singh", 4 -> "Kumar")),
+      Row(Map(10 -> "Nallaa", 20 -> "Sissngh", 100 -> "Gusspta", 40 -> "Kumar"))
+    ))
+  }
+
+  test("Sort Column table property blocking for Map type") {
+    sql("DROP TABLE IF EXISTS carbon")
```
Comment (Indhumathi27) [inline integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/createTable/TestCreateDDLForComplexMapType.scala]: Can you please add some testcases for compaction, partition, preaggregate, IUD and alter scenarios

File: integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/createTable/TestCreateDDLForComplexMapType.scala [hunk 2]
```diff
@@ -0,0 +1,330 @@
...
+           | STORED BY 'carbondata'
+           | TBLPROPERTIES('SORT_COLUMNS'='mapField')
+           | """
+          .stripMargin)
+    }
+    assert(exception1.getMessage
+      .contains(
+        "sort_columns is unsupported for map datatype column: mapfield"))
+  }
+
```
Comment (Indhumathi27) [inline integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/createTable/TestCreateDDLForComplexMapType.scala]: please add a test case to validate empty record as map value
